### PR TITLE
feat: add vite plugin

### PR DESCRIPTION
**Pull request by bigopon**
# Pull Request

## 📖 Description

- Add a plugin to support Vite
- Refactor hmr generation code, so that different plugin can generate hmr code differently, also independently with existing compiled component code. Example:

```
import { CustomElement as $$CE } from '@aurelia/runtime-html';
```

### CODE REVIEW

File: packages/__e2e__/hmr-vite/vite.config.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        path: id,
+        contents: code,
+      }, {
+        // hmr: true,
+        hmrModule: 'import.meta',
+        getHmrCode,
+        transformHtmlImportSpecifier: (s) => {
+          return this.meta.watchMode
+            ? s
+            : s.replace(/\.html$/, '.$au.ts')
```
Comment (bigopon) [inline packages/__e2e__/hmr-vite/vite.config.js:60]: @3cp @brandonseydel  This the way the plugin works, to work around issues of vite plugin order difference between dev and production build + watch/non watch modes.
It changes the `.html` import to `.$au.ts` and the intercepting loader from this plugin will take over.

File: packages/__e2e__/hmr-vite/vite.config.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  const code =
+`import { Metadata as $$M } from '@aurelia/metadata';
+import { ExpressionKind as $$EK } from '@aurelia/runtime';
+import {
+  Controller as $$C,
+  CustomElement as $$CE,
+  IHydrationContext as $$IHC,
+  PropertyBinding as $$PB,
+  ContentBinding as $$CB,
+} from '@aurelia/runtime-html';
```
Comment (bigopon) [inline packages/__e2e__/hmr-vite/vite.config.js:111]: with these aliases, the imports for hmr can stay independent of the component import without having to worry about conflicts.

**Comment by 3cp** [comment]
Cheers!
Did you commit the vite package files? Cannot find any.

**Comment by bigopon** [comment]
> Did you commit the vite package files? Cannot find any.

What files do you expect? I'm a bit confused, can you help point out clearer?

**Comment by 3cp** [comment]
I didn't see any new package under packages-tooling.

**Comment by 3cp** [comment]
NVW, I found yours in vite.config.js.

I was expecting you put that logic in a package.

**Pull request merged**